core/data/tests/can_generate_generic_enum/output.py

```diff
@@ -0,0 +1,94 @@
+"""
+ Generated by typeshare 1.11.0
+"""
+from __future__ import annotations
+
+from pydantic import BaseModel, Enum
+from pydantic.generics import GenericModel
+from typing import Dict, Generic, List, Literal, TypeVar, Union
+
+A = TypeVar("A")
+B = TypeVar("B")
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+class GenericEnumTypes(str, Enum):
+    VARIANT_A = "VariantA"
+    VARIANT_B = "VariantB"
+
+class GenericEnumVariantA(GenericModel, Generic[A]):
+    content: A
+
+class GenericEnumVariantB(GenericModel, Generic[B]):
+    content: B
+
+class GenericEnum(BaseModel):
+    model_config = ConfigDict(use_enum_values=True)
+    type: GenericEnumTypes
+    content: Union[GenericEnumVariantA, GenericEnumVariantB]
+
+class StructUsingGenericEnum(BaseModel):
+    enum_field: GenericEnum[str, int]
+
+
+class GenericEnumUsingGenericEnumTypes(str, Enum):
+    VARIANT_C = "VariantC"
+    VARIANT_D = "VariantD"
+    VARIANT_E = "VariantE"
+
+class GenericEnumUsingGenericEnumVariantC(GenericModel, Generic[T]):
+    content: GenericEnum[T, T]
+
+class GenericEnumUsingGenericEnumVariantD(GenericModel, Generic[T]):
+    content: GenericEnum[str, Dict[str, T]]
+
+class GenericEnumUsingGenericEnumVariantE(BaseModel):
+    content: GenericEnum[str, int]
+
+class GenericEnumUsingGenericEnum(BaseModel):
+    model_config = ConfigDict(use_enum_values=True)
+    type: GenericEnumUsingGenericEnumTypes
+    content: Union[GenericEnumUsingGenericEnumVariantC, GenericEnumUsingGenericEnumVariantD, GenericEnumUsingGenericEnumVariantE]
+
+class GenericEnumsUsingStructVariantsVariantF(GenericModel, Generic[T]):
+    """
+    Generated type representing the anonymous struct variant `VariantF` of the `GenericEnumsUsingStructVariants` Rust enum
+    """
+    action: T
+
+
+class GenericEnumsUsingStructVariantsVariantG(GenericModel, Generic[T, U]):
+    """
+    Generated type representing the anonymous struct variant `VariantG` of the `GenericEnumsUsingStructVariants` Rust enum
+    """
+    action: T
+    response: U
+
+
+class GenericEnumsUsingStructVariantsVariantH(BaseModel):
+    """
+    Generated type representing the anonymous struct variant `VariantH` of the `GenericEnumsUsingStructVariants` Rust enum
+    """
+    non_generic: int
+
+
+class GenericEnumsUsingStructVariantsVariantI(GenericModel, Generic[T, U]):
+    """
+    Generated type representing the anonymous struct variant `VariantI` of the `GenericEnumsUsingStructVariants` Rust enum
+    """
+    vec: List[T]
+    action: MyType[T, U]
```

Comment:
For all algebraic enums (beside unit) I think we should add those helper functions I talked about before.